eg/mock_mailer.py

```diff
@@ -34,16 +54,57 @@ class MockMailer(mail.base.Postman):
     """
 
     file_path: T.Path = Path("/tmp/mail")
+    sent_mail: T.Set[_MockEmail] = set()
+
+    @classmethod
+    def clean(cls) -> None:
+        try:
+            os.remove(cls.file_path)
+        except FileNotFoundError:
+            pass
+
+        cls.sent_mail = set()
 
     @property
     def addresser(self) -> str:
         return "fake-emailer@sanger.ac.uk"
 
     def _deliver(self, message: mail.base.Message,
                  recipients: T.Collection[str], sender: str) -> None:
-        with open(self.__class__.file_path, "w") as f:
+
+        self.__class__.sent_mail.add(_MockEmail(
+            subject=message.subject,
+            to=set(recipients),
+            sender=sender,
+            body=message.body,
+            attachments=set(str(attachment.data) if attachment.mime_type != "application/gzip" else gzip.decompress(
+                attachment.data.read()).decode("utf-8") for attachment in message.attachments)
+        ))
+
+        with open(self.__class__.file_path, "a") as f:
             f.write(f"Subject: {message.subject}\n")
             f.write(f"To: {', '.join(recipients)}\n")
             f.write(f"From: {sender}\n")
             f.write("\n")
             f.write(message.body)
+            f.write("\nAttachments:\n")
+            for attachment in message.attachments:
+                f.write(attachment.filename)
+                if attachment.mime_type == "application/gzip":
+                    f.write(str(gzip.decompress(attachment.data.read())))
+                else:
+                    f.write(str(attachment.data))
+            f.write("\n===\n")
+
+    @classmethod
+    def get_sent_mail(cls, subject: T.Optional[str] = None,
+                      recipient: T.Optional[str] = None) -> T.Set[_MockEmail]:
+        emails = cls.sent_mail
+
+        if subject:
+            emails = {x for x in emails if x.subject == subject}
+
+        if recipient:
+            emails = {x for x in emails if recipient in x.to}
```

Comment:
If you didn't use this in your tests, does it need to be here?

Reply:
i wrote it cause i may have used it